src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -561,6 +573,21 @@ private Task FetchChildren()
                         i++;
                     }
                 }
+                else if (IsMapType())
+                {
+                    //
+                    // support for gdb's pretty-printing built-in displayHint "map"
+                    //
+                    List<VariableInformation> listChildren = new List<VariableInformation>();
+                    for(int p = 0; p+1 < children.Length; p+=2)
```

Comment:
Yeah, this is quite confusing for someone with no experience with these mi records.